### PR TITLE
Remove BuildResult - CLOUDBLD-8975

### PR DESCRIPTION
**Pull request by tkdchen**
Detailed description is in the individual commit message.

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Python type annotations added to new code
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by tkdchen** [comment]
Note: This pr is related to #1843. If it gets merged firstly, rebase this pr. For the group_manifests.

**Comment by tkdchen** [comment]
Rebased.

File: atomic_reactor/plugins/build_source_container.py [hunk 1]
```diff
@@ -150,13 +150,7 @@ def run(self):
                           maven_source_data_dir)
             shutil.rmtree(maven_source_data_dir)
 
-        image_tar_path = self.export_image(image_output_dir)
+        self.export_image(image_output_dir)
 
         self.log.info('Will remove unpacked image directory: %s', image_output_dir)
         shutil.rmtree(image_output_dir)
-
```
Comment (rcerven) [inline atomic_reactor/plugins/build_source_container.py:157]: well we shouldn't return BuildResult anymore BUT, we should return the same as dict .....

File: atomic_reactor/plugins/build_source_container.py [hunk 2]
```diff
@@ -46,7 +46,6 @@ def export_image(self, image_output_dir: Path) -> str:
         img_metadata = get_exported_image_metadata(str(output_path), IMAGE_TYPE_DOCKER_ARCHIVE)
         # OSBS2 TBD exported_image_sequence will not work for multiple platform
         self.workflow.data.exported_image_sequence.append(img_metadata)
-        return str(output_path)
```
Comment (rcerven) [inline atomic_reactor/plugins/build_source_container.py:49]: we still want that

Comment (rcerven) [inline atomic_reactor/plugins/build_source_container.py:49]: we still want that

File: atomic_reactor/plugins/build_source_container.py [hunk 3]
```diff
@@ -150,13 +150,7 @@ def run(self):
                           maven_source_data_dir)
             shutil.rmtree(maven_source_data_dir)
 
-        image_tar_path = self.export_image(image_output_dir)
+        self.export_image(image_output_dir)
 
         self.log.info('Will remove unpacked image directory: %s', image_output_dir)
         shutil.rmtree(image_output_dir)
-
-        return BuildResult(
```
Comment (rcerven) [inline atomic_reactor/plugins/build_source_container.py:158]: well we shouldn't return BuildResult anymore BUT, we should return the same as dict .....

**Review by rcerven — CHANGES_REQUESTED** [review]

File: atomic_reactor/plugins/build_source_container.py (hunk 2 above)
Comment (tkdchen) [reply atomic_reactor/plugins/build_source_container.py:49]: @rcerven Could you be specific why we want this? The source docker archive can be accessible from the `BuildDir.exported_squashed_image`.

File: atomic_reactor/plugins/build_source_container.py (hunk 3 above)
Comment (chmeliik) [reply atomic_reactor/plugins/build_source_container.py:158]: Hmm, and why do we want that?

**Comment by chmeliik** [comment]
Mostly LGTM

Should we also remove build result handling from the ContextDir?

File: atomic_reactor/plugins/build_source_container.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/build_source_container.py:158]: ahh right so we know that output dir already in advance, right so logs and skip_layer as return looks good

File: atomic_reactor/plugins/build_source_container.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/build_source_container.py:49]: right that will do

**Review by rcerven — APPROVED** [review]

**Comment by tkdchen** [comment]
@chmeliik Updated. PTAL.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Pull request merged**